pallets/pallet-web3-names/src/lib.rs

```diff
@@ -0,0 +1,482 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2022 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+//! # Pallet storing unique nickname <-> DID links for user-friendly DID
+//! nicknames.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod web3_name;
+
+mod default_weights;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+pub use crate::{default_weights::WeightInfo, pallet::*};
+
+#[frame_support::pallet]
+pub mod pallet {
+	use codec::FullCodec;
+	use frame_support::{
+		pallet_prelude::*,
+		sp_runtime::SaturatedConversion,
+		traits::{Currency, ReservableCurrency, StorageVersion},
+		Blake2_128Concat,
+	};
+	use frame_system::pallet_prelude::*;
+	use sp_std::{fmt::Debug, vec::Vec};
+
+	use kilt_support::{deposit::Deposit, traits::CallSources};
+
+	use super::WeightInfo;
+	use crate::web3_name::Web3NameOwnership;
+
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+	pub type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+	pub type BalanceOf<T> = <<T as Config>::Currency as Currency<AccountIdOf<T>>>::Balance;
+	pub type BlockNumberFor<T> = <T as frame_system::Config>::BlockNumber;
+	pub type Web3NameOwnerOf<T> = <T as Config>::Web3NameOwner;
+	pub type Web3NameInput<T> = BoundedVec<u8, <T as Config>::MaxNameLength>;
+	pub type Web3NameOf<T> = <T as Config>::Web3Name;
+	pub type Web3OwnershipOf<T> =
+		Web3NameOwnership<Web3NameOwnerOf<T>, Deposit<AccountIdOf<T>, BalanceOf<T>>, BlockNumberFor<T>>;
+
+	pub(crate) type CurrencyOf<T> = <T as Config>::Currency;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(_);
+
+	// Web3Name -> Ownership
```

Comment:
maybe those comments could be documentation?
```suggestion
	/// Web3Name -> Ownership
```

Reply:
https://github.com/KILTprotocol/mashnet-node/pull/314/commits/53aa550ad7fec86b420dd9eaf574f9dc3aba5ad8